File: typescript/leetcode/lib/List.ts
export interface List<T> {
  val: T
  next: List<T> | null
}

export function createList<T>(values: T[]): List<T> | null {
  if (values.length === 0) {
    return null
  }

  let node: List<T> | null = null

  for (let i = values.length - 1; i >= 0; i--) {
    const prev: List<T> = { val: values[i], next: null }
    prev.next = node
    node = prev
  }

  return node
}

export function toArray<T>(head: List<T> | null): T[] {
  const values = []
  while (head) {
    values.push(head.val)
    head = head.next
  }
  return values
}

File: python/leetcode/lib/list.py
from typing import Generic, TypeVar, Optional, List

T = TypeVar("T")


class ListNode(Generic[T]):
    val: T
    next: Optional["ListNode[T]"]

    def __init__(self, val: T):
        self.val = val
        self.next = None


def create_linked_list(values: List[T]) -> Optional[ListNode[T]]:
    """Create a linked-list into a list

    Arguments:
        values {List[T]} -- values of list

    Returns:
        Optional[ListNode[T]] -- head of the linked-list
    """
    node = None
    for val in reversed(values):
        prev = ListNode(val)
        prev.next = node
        node = prev

    return node


def create_list(head: Optional[ListNode[T]]) -> List[T]:
    """Convert a linked-list into a list

    Arguments:
        head {Optional[ListNode[T]]} -- The head of linked-list

    Returns:
        List[T] -- list of values of linked-list
    """
    values: List[T] = []

    while head:
        values.append(head.val)
        head = head.next

    return values
